src/base.jl

```diff
@@ -23,25 +23,28 @@ function issquare(A::AbstractMatrix)
     return m == n
 end
 
-abstract type AbstractSquareKronecker <: AbstractKroneckerProduct end
+"""
+    issquare(K::AbstractKroneckerProduct)
 
-# general Kronecker product between two square matrices
-struct SquareKroneckerProduct{T, TA<:AbstractMatrix, TB<:AbstractMatrix} <: AbstractSquareKronecker
-    A::TA
-    B::TB
-    function SquareKroneckerProduct(A::AbstractMatrix{T}, B::AbstractMatrix{V}) where {T, V}
-        if issquare(A) && issquare(B)
-            return new{promote_type(T, V), typeof(A), typeof(B)}(A, B)
-        else
-            throw(DimensionMismatch(
-                "SquareKroneckerProduct is only for when all matrices are square",
-            ))
-        end
-    end
+Checks if all matrices of a Kronecker product are square.
+"""
+function issquare(K::AbstractKroneckerProduct)
+    return issquare(K.A) && issquare(K.B)
```

Comment:
Can be inlined to `issquare(K::AbstractKroneckerProduct) = issquare(K.A) && issquare(K.B)`